### PR TITLE
Decimal arithmetic operation inspection

### PR DESCRIPTION
**Pull request by lrubasze**
## Summary
Inspect Decimal operations unwraps.
- replace (if necessary) them with proper error handling
- annotate those safe ones
- annotate those to be done later

NOTE: Unwraps in tests were left as is.

### CODE REVIEW

File: radix-engine/src/blueprints/consensus_manager/validator.rs [hunk 1]
```diff
@@ -1589,13 +1612,13 @@ impl ValidatorBlueprint {
             Decimal::zero()
         } else {
             amount_of_stake_units
-                .safe_div(total_stake_unit_supply)
-                .and_then(|d| d.safe_mul(active_stake_amount))
-                .ok_or_else(|| {
-                    RuntimeError::ApplicationError(ApplicationError::ValidatorError(
+                .safe_mul(active_stake_amount)
```
Comment (talekhinezh) [inline radix-engine/src/blueprints/consensus_manager/validator.rs:1615]: We should do division before multiplication here to reduce chance of overflow as well as be more consistent with pools

File: radix-engine/src/blueprints/consensus_manager/validator.rs [hunk 2]
```diff
@@ -1607,20 +1630,18 @@ impl ValidatorBlueprint {
...
-            xrd_amount
+        if total_stake_xrd_amount.is_zero() {
+            Ok(xrd_amount)
         } else {
             xrd_amount
-                .safe_div(total_stake_xrd_amount)
-                .and_then(|d| d.safe_mul(total_stake_unit_supply))
-                .ok_or_else(|| {
-                    RuntimeError::ApplicationError(ApplicationError::ValidatorError(
+                .safe_mul(total_stake_unit_supply)
```
Comment (talekhinezh) [inline radix-engine/src/blueprints/consensus_manager/validator.rs:1637]: We should do division before multiplication here to reduce chance of overflow as well as be more consistent with pools

File: radix-engine/src/blueprints/consensus_manager/validator.rs (hunk 1 above)
Comment (lrubasze) [reply radix-engine/src/blueprints/consensus_manager/validator.rs:1615]: good catch!

**Comment by 0xOmarA** [comment]
I found a `safe_sub().unwrap` here as well: https://github.com/radixdlt/radixdlt-scrypto/blob/fcd1b2d235a79bdb13790d2854ee6f54df4f3141/radix-engine/src/blueprints/resource/non_fungible/non_fungible_vault.rs#L419

File: radix-engine/src/blueprints/consensus_manager/validator.rs (hunk 1 above)
Comment (lrubasze) [reply radix-engine/src/blueprints/consensus_manager/validator.rs]: Done

**Comment by lrubasze** [comment]
@0xOmarA @talekhinezh found also these ones
- `put()` function
https://github.com/radixdlt/radixdlt-scrypto/blob/fcd1b2d235a79bdb13790d2854ee6f54df4f3141/radix-engine-interface/src/blueprints/resource/resource.rs#L53
- `take_by_amount()` function
https://github.com/radixdlt/radixdlt-scrypto/blob/fcd1b2d235a79bdb13790d2854ee6f54df4f3141/radix-engine-interface/src/blueprints/resource/resource.rs#L64
Is it fine to leave it as is?
`take_by_amount()` already returns an error, so this one is easy.
But `put()` does not, additionally it is used in many places (eg. `transaction_executor()`)

**Review by talekhinezh — DISMISSED** [review]

**Review by talekhinezh — APPROVED** [review]

**Pull request merged**